### ISSUE TITLE
block-scoped-var false positives on some export constructions

### ISSUE THREAD

**Issue by mathieumg** [eslint/eslint#2887]
The following trigger the "used outside of binding context" error/warning for the `block-scoped-var` rule even though they shouldn't:

```js
export {a} from "mod";
export {a as b} from "mod";
```

This is different from #2807 which is a construct from the experimental https://github.com/leebyron/ecmascript-more-export-from
